packages/sanity/src/_internal/cli/actions/backup/resolveApiClient.ts

```diff
@@ -0,0 +1,45 @@
+import {CliCommandContext} from '@sanity/cli'
+import {SanityClient} from '@sanity/client'
+import {chooseDatasetPrompt} from '../dataset/chooseDatasetPrompt'
+
+type ResolvedApiClient = {
+  projectId: string
+  datasetName: string
+  token?: string
+  client: SanityClient
+}
+
+async function resolveApiClient(
+  context: CliCommandContext,
+  datasetName: string,
+  apiVersion: string,
+): Promise<ResolvedApiClient> {
+  const {apiClient} = context
+
+  let client = apiClient()
+  const {projectId, token} = client.config()
+
+  if (!projectId) {
+    throw new Error('Project ID not defined')
+  }
+
+  // If no dataset provided, explicitly ask for dataset instead of using dataset
+  // configured in Sanity config. Aligns with `sanity dataset export` behavior.
+  let selectedDataset: string = datasetName
+  if (!selectedDataset) {
+    selectedDataset = await chooseDatasetPrompt(context, {
+      message: 'Select the dataset name:',
+    })
+  }
+
+  client = client.clone().config({dataset: datasetName, apiVersion})
```

Comment:
probably not a big deal but I believe `client.withConfig({ dataset: datasetName, apiVersion})` does the same thing

Reply:
Replaced it with `withConfig`.